Add tutorial section for documentation (#6213)
holoviz/panel: doc/explanation/develop_seamlessly.md

> # Develop seamlessly Across Environments

Reviewer: ```suggestion
# Develop Seamlessly Across Environments
```

Or sentence case.

> We believe you and your team develop in a wide range of development environments. For example in a notebook for data exploration, model training, research or study. For example in an editor for other use cases. That means you and your team will be utilizing a combination of development environments like Jupyter Notebook, Jupyter Lab, Jupyter Hub, Binder, Google Colab, Spyder, VS Code, PyCharm etc..

Reviewer: Personally, I think we should put the main point first before the user gets lost in this text.

```suggestion
To facilitate seamless development across a diverse array of environments, we aim to empower you and your team with the capability to utilize Panel effectively. This entails leveraging various development environments such as Jupyter Notebook, Jupyter Lab, Jupyter Hub, Binder, Google Colab, Spyder, VS Code, PyCharm, and more.
```

> Therefore **we want to enable you and your team to develop seamlessly with Panel across a wide range of development environments**.

Reviewer: Combined above.
```suggestion
```

## Notebook

We love notebooks.

### Notebook friendly

> Jupyter notebooks have been supported in Panel since day 1 and more generally since the inception of the [HoloViz](https://holoviz.org) project. Notebooks are environments that allow to iterate quickly, running some code and observing instantaneously its output. Notebooks also happen to be a valid deployment option:

Reviewer: Not sure if need to mention "more generally since the inception"
```suggestion
Jupyter notebooks have been supported in Panel since day 1. Notebooks are environments that allow one to iterate quickly, running some code and observing instantaneously its output. Notebooks also happen to be a valid deployment option:
```
